x509/x509.go

```diff
@@ -1183,6 +1188,23 @@ type distributionPointName struct {
 	RelativeName pkix.RDNSequence `asn1:"optional,tag:1"`
 }
 
+// TorServiceDescriptorHash is a structure corrsponding to the
+// TorServiceDescriptorHash SEQUENCE described in Appendix F ("Issuance of
+// Certificates for .onion Domain Names").
+//
+// Each TorServiceDescriptorHash holds an onion URI (a utf8 string with the
+// .onion address that was validated), a hash algorithm name (computed based on
+// the pkix.AlgorithmIdentifier in the TorServiceDescriptorHash), the hash bytes
+// (computed over the DER encoding of the ASN.1 SubjectPublicKey of the .onion
+// service), and the number of bits in the hash bytes.
+type TorServiceDescriptorHash struct {
+	Onion         string                   `json:"onion"`
+	Algorithm     pkix.AlgorithmIdentifier `json:"-"`
+	AlgorithmName string                   `json:"algorithm_name"`
+	Hash          []byte                   `json:"hash"`
```

Comment:
We usually output hashes as hex, not Base64. There's an alias type somewhere in ZCrypto (maybe called `CertificateFingerprint`?) that does this transparently. I'd update the JSON serialization and/or type definition of `TorServiceDescriptorHash` to use a `Fingerprint` type.

Reply:
Sounds good, can do.

Reply:
Done as of a7bcfac :ballot_box_with_check: